### PR TITLE
Add Admin::Trends methods

### LINKED ISSUE #326: Implement Admin Trends API
Mastodon's Admin Trends API allows an admin to view trending links, statuses and tags. The goal of this task is to implement the following endpoints of this API in BigBone:

- View trending links
- View trending statuses
- View trending tags

The Admin Reports API as it is documented by Mastodon can be found here:
https://docs.joinmastodon.org/methods/admin/trends/

Tip: A lot of other endpoints similar to this have already been implemented. In order to get an idea how to implement this, I suggest that you study the code upfront. Start with one of the `*Methods.kt` files.

Please consult the following documents to make sure your contribution meets our projects criteria:

- [Code of Conduct](https://github.com/andregasser/bigbone/blob/master/CODE_OF_CONDUCT.md)
- [Code Contribution Guidelines](https://github.com/andregasser/bigbone/blob/master/CONTRIBUTING.md)

### PR DESCRIPTION
**Pull request by PattaFeuFeu**
# Description

Adds methods for Admins to get trending tags, statuses, and links.

> [!NOTE]
> We don’t have a good system yet for entities that have similar versions for admin- and non-admin variants, like in this case with `Tag`. `AdminTag` shares most of its properties with `Tag` but adds some additional ones.
> For now, I have just added a new entity with copies from `Tag` and added the missing ones to it.

Closes #326 

# Type of Change

- New feature
- Documentation

# Breaking Changes

- None

# How Has This Been Tested?

New unit tests for all three methods

# Mandatory Checklist

- [x] I ran `gradle check` and there were no errors reported
- [x] I have performed a self-review of my code
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] All tests pass locally with my changes
- [x] I have added KDoc documentation to all public methods

# Optional Things To Check

The items below are some more things to check before asking other people to review your code.

- [x] In case you worked on a new feature: Did you also implement the reactive endpoint (bigbone-rx)?
- [x] In case you added new `*Methods` classes: Did you also reference it in the `MastodonClient` main class?
- [x] Did you also update the documentation in the `/docs` folder (e.g. API Coverage page)?

### CODE REVIEW

File: bigbone-rx/src/main/kotlin/social/bigbone/rx/admin/RxAdminTrendMethods.kt [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+    private val adminTrendMethods = AdminTrendMethods(client)
+
+    /**
+     * Links that have been shared more than others, including unapproved and unreviewed links.
+     *
+     * @see <a href="https://docs.joinmastodon.org/methods/admin/trends/#links">Mastodon API documentation: admin/trends/#links</a>
+     */
+    fun getTrendingLinks(): Single<MastodonRequest<List<TrendsLink>>> = Single.fromCallable {
+        adminTrendMethods.getTrendingLinks()
```
Comment (andregasser) [inline bigbone-rx/src/main/kotlin/social/bigbone/rx/admin/RxAdminTrendMethods.kt:28]: I think the `.execute()` call is missing here (and in the other methods below).

Comment (PattaFeuFeu) [reply bigbone-rx/src/main/kotlin/social/bigbone/rx/admin/RxAdminTrendMethods.kt]: Whoops! Fixed. 😊

**Review by andregasser — CHANGES_REQUESTED** [review]
Just one finding in `RxAdminTrendMethods.kt`. Otherwise looking good!

**Review by andregasser — APPROVED** [review]
👍

**Pull request merged**